src/main/java/com/amazonaws/encryptionsdk/model/DecryptionMaterials.java

```diff
@@ -37,24 +78,47 @@ public Builder toBuilder() {
     }
 
     public static final class Builder {
-        private DataKey<?> dataKey;
+        private CryptoAlgorithm algorithm;
+        private SecretKey cleartextDataKey;
+        private MasterKey<?> masterKey;
         private PublicKey trailingSignatureKey;
+        private Map<String, String> encryptionContext;
         private KeyringTrace keyringTrace;
 
         private Builder(DecryptionMaterials result) {
-            this.dataKey = result.getDataKey();
+            this.algorithm = result.getAlgorithm();
+            this.cleartextDataKey = result.getCleartextDataKey();
+            this.masterKey = result.getMasterKey();
             this.trailingSignatureKey = result.getTrailingSignatureKey();
             this.keyringTrace = result.getKeyringTrace();
         }
 
         private Builder() {}
 
-        public DataKey<?> getDataKey() {
-            return dataKey;
+        public CryptoAlgorithm getAlgorithm() {
+            return algorithm;
+        }
+
+        public Builder setAlgorithm(CryptoAlgorithm algorithm) {
+            this.algorithm = algorithm;
+            return this;
+        }
+
+        public SecretKey getCleartextDataKey() {
+            return cleartextDataKey;
```

Comment:
Not a java expert :) . Can someone who gets this value manipulate it and change it for subsequent callers?  Or is this `final`?

Reply:
SecretKeySpec, which implements the SecretKey interface, clones the encoded key byte array before returning it, so it should be ok:

```java
public byte[] getEncoded() {
    return (byte[])this.key.clone();
}
```